lib/mayaUsd/ufe/UsdMaterial.h

```diff
@@ -0,0 +1,50 @@
+// ===========================================================================
+// Copyright 2022 Autodesk, Inc. All rights reserved.
+//
+// Use of this software is subject to the terms of the Autodesk license
+// agreement provided at the time of installation or download, or which
+// otherwise accompanies this software in either electronic or hard copy form.
+// ===========================================================================
+#pragma once
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+#include <ufe/material.h>
+#include <ufe/sceneItem.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief USD run-time material interface
+/*!
+    This class implements the Material interface for USD prims.
+*/
+class MAYAUSD_CORE_PUBLIC UsdMaterial : public Ufe::Material
+{
+public:
+    using Ptr = std::shared_ptr<UsdMaterial>;
+
+    UsdMaterial(const UsdSceneItem::Ptr& item);
+    ~UsdMaterial() override;
+
+    // Delete the copy/move constructors assignment operators.
+    UsdMaterial(const UsdMaterial&) = delete;
+    UsdMaterial& operator=(const UsdMaterial&) = delete;
+    UsdMaterial(UsdMaterial&&) = delete;
+    UsdMaterial& operator=(UsdMaterial&&) = delete;
+
+    //! Create a UsdMaterial.
+    static UsdMaterial::Ptr create(const UsdSceneItem::Ptr& item);
+
+    std::vector<Ufe::SceneItem::Ptr> getMaterials() const override;
+
+private:
+    std::vector<PXR_NS::UsdPrim> getConnectedShaders(const PXR_NS::UsdShadeOutput& port) const;
+
+private:
+    UsdSceneItem::Ptr fItem;
```

Comment:
The [coding standard](https://github.com/Autodesk/maya-usd/blob/dev/doc/codingGuidelines.md) for MayaUsd is to use _item